### ISSUE TITLE
There was a crash when  invoke this method '- (void)sd_setImageWithURL: placeholderImage:completed:' on main queue

### ISSUE THREAD

**Issue by zhenyiyi** [rs/SDWebImage#1871]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 8.2
 SDWebImage Version      |  3.8.2
 Integration Method      |cocoapods
 Xcode Version           |  Xcode 8.1
 Repro rate              |  only once
 Repro with our demo prj | 
 Demo project link       | 

There was a crash when  invoke this method '- (void)sd_setImageWithURL: placeholderImage:completed:' on main queue. It only happens once.
![1492600133847_0](https://cloud.githubusercontent.com/assets/8273338/25320274/0fe4e38e-28d8-11e7-86de-b3b132e75192.jpg)

**Comment by mythodeia** [comment]
can you try with the latest version? can you replicate it on the demo project?

**Comment by ashishprasadgit** [comment]
I also faced similar crash
`Crashed: com.apple.main-thread
0  libobjc.A.dylib                0x18bef0150 objc_msgSend + 16
1  libobjc.A.dylib                0x18beed57c _object_set_associative_reference + 580
2  SDWebImage                     0x1032f6724 -[UIView(WebCache) sd_internalSetImageWithURL:placeholderImage:options:operationKey:setImageBlock:progress:completed:] (UIView+WebCache.m:39)
3  SDWebImage                     0x1032f5d94 -[UIImageView(WebCache) sd_setImageWithURL:placeholderImage:options:progress:completed:] (UIImageView+WebCache.m:48)
4  SDWebImage                     0x1032f5c7c -[UIImageView(WebCache) sd_setImageWithURL:placeholderImage:completed:] (UIImageView+WebCache.m:36)
**5  - 8 - Removed Code related functions**
8  UIKit                          0x193ea4268 -[UICollectionView _createPreparedCellForItemAtIndexPath:withLayoutAttributes:applyAttributes:isFocused:notify:] + 448
9  UIKit                          0x19360a884 -[UICollectionView _updateVisibleCellsNow:] + 3820
10 UIKit                          0x1936054ac -[UICollectionView layoutSubviews] + 300
11 UIKit                          0x1935a7158 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1200
12 QuartzCore                     0x190797274 -[CALayer layoutSublayers] + 148
13 QuartzCore                     0x19078bde8 CA::Layer::layout_if_needed(CA::Transaction*) + 292
14 QuartzCore                     0x19078bca8 CA::Layer::layout_and_display_if_needed(CA::Transaction*) + 32
15 QuartzCore                     0x190707360 CA::Context::commit_transaction(CA::Transaction*) + 252
16 QuartzCore                     0x19072e3c0 CA::Transaction::commit() + 504
17 QuartzCore                     0x19072ee8c CA::Transaction::observer_callback(__CFRunLoopObserver*, unsigned long, void*) + 120
18 CoreFoundation                 0x18d4289a0 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 32
19 CoreFoundation                 0x18d426628 __CFRunLoopDoObservers + 372
20 CoreFoundation                 0x18d426a74 __CFRunLoopRun + 956
21 CoreFoundation                 0x18d356d94 CFRunLoopRunSpecific + 424
22 GraphicsServices               0x18edc0074 GSEventRunModal + 100
23 UIKit                          0x19360f130 UIApplicationMain + 208
24  -                              0x10013e0f0 main (main.m:14)
25 libdyld.dylib                  0x18c36559c start + 4`

**Comment by mythodeia** [comment]
did you try to replicate this in the demo project? can you show some code? i will close this. please reopen if you have more info
